tests/jballerina-unit-test/src/test/resources/test-src/statements/transactionstatement/trx-stmt-negative.bal

```diff
@@ -0,0 +1,283 @@
+        }
+    }
+}
+
+function testNestedTransactions2() returns error? {
+    transaction {
+        do {
+            transaction {
+                check getError();
+                check commit;
+            }
+        } on fail {
+            rollback;
+        }
+        check commit; // Error: commit not allowed here
+    }
+}
+
+function testMultipleCheckExpressions() returns error? {
+    transaction {
+        do {
+            check getError();
+            check getError();
+            check commit;
+        } on fail {
+            rollback;
+        }
+    }
+}
+
+function testPreCommitExitWithNestedTransaction() returns error? {
+    transaction {
+        transaction { // Error: exit from transaction prior to commit/rollback identified
+            do {
+                check getError();
+                check commit;
+            } on fail {
+            }
+        }
+        check commit;
+    }
+}
+
+function testNestedTransactionRollback() returns error? {
+    transaction {
+        transaction {
+            do {
+                check getError();
+                check commit;
+            } on fail {
+                rollback;
+            }
+        }
+        check commit;
+    }
+}
+
+function testPreCommitExitNoRollback() returns error? {
+    transaction {
+        transaction { // Error: exit from transaction prior to commit/rollback identified
+            do {
+                check getError();
+                check commit;
+            } on fail {
+                // No rollback here
+            }
+        }
+        check commit;
+    }
+}
+
+function testMultiplePreCommitExits1() returns error? {
+    transaction { // Error: exit from transaction prior to commit/rollback identified
+        do {
+            check getError();
+            check commit;
+        } on fail {
+            // No rollback here
+        }
+        do {
+            check getError();
+            check commit; // Error: commit not allowed
+        } on fail {
+            // No rollback here
+        }
+    }
+}
+
+function testCommittedExit5() returns error? {
+    transaction {
+        do {
+            check getError();
+            check commit;
+        } on fail {
+            // No rollback here
+        }
+        check commit; // Error: commit not allowed
+    }
+}
+
+function testMultiplePreCommitExits2() returns error? {
+    transaction {
+        do {
+            check getError();
+            check commit;
+        } on fail {
+            // No rollback here
+        }
+        do {
+            check getError();
+            check commit; // Error: commit not allowed
+        } on fail {
+            rollback;
+        }
+    }
+}
+
+function testPreCommitExit4(boolean foo) returns error? {
+    transaction { // Error: exit from transaction prior to commit/rollback identified
+      if foo {
+        check commit;
+        return;
+      }
+    }
+}
+
+function testCommittedExit6(boolean foo) returns error? {
```

Comment:
Shall we also add few tests related to match statement inside the transaction block?

```bal
function foo() returns error? {
    int|string ss = 56;
    transaction {
        match ss {
            "FD" => {
                check commit;
            }
        }
    }
}
```

This can also apply to multiple if statements as well.